recipes/clipper2/all/conanfile.py

```diff
@@ -0,0 +1,107 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import get, copy, rmdir, export_conandata_patches, apply_conandata_patches
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+import os
+
+required_conan_version = ">=1.53.0"
+
+class Clipper2Conan(ConanFile):
+    name = "clipper2"
+    description = " A Polygon Clipping and Offsetting library in C++"
+    license = "BSL-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/AngusJohnson/Clipper2"
+    topics = ("geometry", "polygon", "clipping")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "usingz": ["ON", "OFF", "ONLY"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "usingz": "ON",
```

Comment:
Can we get a log of this working OFF/ONLY working ok? Thanks! :)

Reply:
@RubenRBS 

Please check following:

```
> conan create . -o clipper2/\*:usingz=ON --build=missing --version=1.2.2

======== Exporting recipe to the cache ========
clipper2/1.2.2: Exporting package recipe: /home/toge/src/conan-center-index/recipes/clipper2/all/conanfile.py
clipper2/1.2.2: exports: File 'conandata.yml' found. Exporting it...
clipper2/1.2.2: Calling export_sources()
clipper2/1.2.2: Copied 1 '.yml' file: conandata.yml
clipper2/1.2.2: Copied 1 '.py' file: conanfile.py
clipper2/1.2.2: Copied 1 '.patch' file: 1.2.2-0001-fix-install.patch
clipper2/1.2.2: Exported to cache folder: /home/toge/.conan2/p/clippd442525f1db72/e
clipper2/1.2.2: Exported: clipper2/1.2.2#7725035548b161d11632f2647eee549b (2023-07-17 15:17:15 UTC)

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=20
compiler.libcxx=libstdc++11
compiler.version=13
os=Linux
[options]
clipper2/*:usingz=ON

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=20
compiler.libcxx=libstdc++11
compiler.version=13
os=Linux


======== Computing dependency graph ========
Graph root
    cli
Requirements
    clipper2/1.2.2#7725035548b161d11632f2647eee549b - Cache

======== Computing necessary packages ========
clipper2/1.2.2: Checking 5 compatible configurations:
clipper2/1.2.2: '65b4f26a1adf5c26829123cc1b2b0c6fb2f6c124': compiler.cppstd=17
clipper2/1.2.2: 'ac9c54fd37c67f97529d2eb5b041569d0f0df94f': compiler.cppstd=gnu17
clipper2/1.2.2: '993634dd15dc627f7c49fe8393074fa4bba8c4e2': compiler.cppstd=gnu20
clipper2/1.2.2: '9978daf07241e68b6b1e00f1456313544734f74a': compiler.cppstd=23
clipper2/1.2.2: '2b36700c1ec7691d94d3ad6363a78cd66329c4f5': compiler.cppstd=gnu23
Requirements
    clipper2/1.2.2#7725035548b161d11632f2647eee549b:38b07c1e6edb13b4efbe474d797049f16efb48bc - Build

======== Installing packages ========
clipper2/1.2.2: Calling source() in /home/toge/.conan2/p/clippd442525f1db72/s/src
clipper2/1.2.2: Unzipping 3.4MB, this can take a while
Unzipping 100 %                                                       

-------- Installing package clipper2/1.2.2 (1 of 1) --------
clipper2/1.2.2: Building from source
clipper2/1.2.2: Package clipper2/1.2.2:38b07c1e6edb13b4efbe474d797049f16efb48bc
clipper2/1.2.2: Copying sources to build folder
clipper2/1.2.2: Building your package in /home/toge/.conan2/p/b/clipp5f826b8dae37d/b
clipper2/1.2.2: Calling generate()
clipper2/1.2.2: Generators folder: /home/toge/.conan2/p/b/clipp5f826b8dae37d/b/build/Release/generators
clipper2/1.2.2: CMakeToolchain generated: conan_toolchain.cmake
clipper2/1.2.2: CMakeToolchain generated: CMakePresets.json
clipper2/1.2.2: Generating aggregated env files
clipper2/1.2.2: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
clipper2/1.2.2: Calling build()
clipper2/1.2.2: Apply patch (portability): fix install for windows shared build
clipper2/1.2.2: Running CMake.configure()
clipper2/1.2.2: RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/home/toge/.conan2/p/b/clipp5f826b8dae37d/b/build/Release/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/home/toge/.conan2/p/b/clipp5f826b8dae37d/p" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/home/toge/.conan2/p/b/clipp5f826b8dae37d/b/src/CPP"
-- Using Conan toolchain: /home/toge/.conan2/p/b/clipp5f826b8dae37d/b/build/Release/generators/conan_toolchain.cmake
-- Conan toolchain: Setting CMAKE_POSITION_INDEPENDENT_CODE=ON (options.fPIC)
-- Conan toolchain: C++ Standard 20 with extensions OFF
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- The C compiler identification is GNU 13.1.1
-- The CXX compiler identification is GNU 13.1.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done (0.2s)
-- Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_POLICY_DEFAULT_CMP0091


-- Build files have been written to: /home/toge/.conan2/p/b/clipp5f826b8dae37d/b/build/Release

clipper2/1.2.2: Running CMake.build()
clipper2/1.2.2: RUN: cmake --build "/home/toge/.conan2/p/b/clipp5f826b8dae37d/b/build/Release" -- -j12
[ 37%] Building CXX object CMakeFiles/Clipper2Z.dir/Clipper2Lib/src/clipper.offset.cpp.o
[ 37%] Building CXX object CMakeFiles/Clipper2.dir/Clipper2Lib/src/clipper.engine.cpp.o
[ 37%] Building CXX object CMakeFiles/Clipper2Z.dir/Clipper2Lib/src/clipper.rectclip.cpp.o
[ 50%] Building CXX object CMakeFiles/Clipper2Z.dir/Clipper2Lib/src/clipper.engine.cpp.o
[ 75%] Building CXX object CMakeFiles/Clipper2.dir/Clipper2Lib/src/clipper.offset.cpp.o
[ 75%] Building CXX object CMakeFiles/Clipper2.dir/Clipper2Lib/src/clipper.rectclip.cpp.o
[ 87%] Linking CXX static library libClipper2.a
[ 87%] Built target Clipper2
[100%] Linking CXX static library libClipper2Z.a
[100%] Built target Clipper2Z

clipper2/1.2.2: Package '38b07c1e6edb13b4efbe474d797049f16efb48bc' built
clipper2/1.2.2: Build folder /home/toge/.conan2/p/b/clipp5f826b8dae37d/b/build/Release
clipper2/1.2.2: Generating the package
clipper2/1.2.2: Temporary package folder /home/toge/.conan2/p/b/clipp5f826b8dae37d/p
clipper2/1.2.2: Calling package()
clipper2/1.2.2: Running CMake.install()
clipper2/1.2.2: RUN: cmake --install "/home/toge/.conan2/p/b/clipp5f826b8dae37d/b/build/Release" --prefix "/home/toge/.conan2/p/b/clipp5f826b8dae37d/p"
-- Install configuration: "Release"
-- Installing: /home/toge/.conan2/p/b/clipp5f826b8dae37d/p/lib/libClipper2.a
-- Installing: /home/toge/.conan2/p/b/clipp5f826b8dae37d/p/include/clipper2/clipper.h
-- Installing: /home/toge/.conan2/p/b/clipp5f826b8dae37d/p/include/clipper2/clipper.core.h
-- Installing: /home/toge/.conan2/p/b/clipp5f826b8dae37d/p/include/clipper2/clipper.engine.h
-- Installing: /home/toge/.conan2/p/b/clipp5f826b8dae37d/p/include/clipper2/clipper.export.h
-- Installing: /home/toge/.conan2/p/b/clipp5f826b8dae37d/p/include/clipper2/clipper.minkowski.h
-- Installing: /home/toge/.conan2/p/b/clipp5f826b8dae37d/p/include/clipper2/clipper.offset.h
-- Installing: /home/toge/.conan2/p/b/clipp5f826b8dae37d/p/include/clipper2/clipper.rectclip.h
-- Installing: /home/toge/.conan2/p/b/clipp5f826b8dae37d/p/lib/libClipper2Z.a
-- Up-to-date: /home/toge/.conan2/p/b/clipp5f826b8dae37d/p/include/clipper2/clipper.h
-- Up-to-date: /home/toge/.conan2/p/b/clipp5f826b8dae37d/p/include/clipper2/clipper.core.h
-- Up-to-date: /home/toge/.conan2/p/b/clipp5f826b8dae37d/p/include/clipper2/clipper.engine.h
-- Up-to-date: /home/toge/.conan2/p/b/clipp5f826b8dae37d/p/include/clipper2/clipper.export.h
-- Up-to-date: /home/toge/.conan2/p/b/clipp5f826b8dae37d/p/include/clipper2/clipper.minkowski.h
-- Up-to-date: /home/toge/.conan2/p/b/clipp5f826b8dae37d/p/include/clipper2/clipper.offset.h
-- Up-to-date: /home/toge/.conan2/p/b/clipp5f826b8dae37d/p/include/clipper2/clipper.rectclip.h
-- Installing: /home/toge/.conan2/p/b/clipp5f826b8dae37d/p/lib/pkgconfig/Clipper2.pc
-- Installing: /home/toge/.conan2/p/b/clipp5f826b8dae37d/p/lib/pkgconfig/Clipper2Z.pc

clipper2/1.2.2: package(): Packaged 1 file: LICENSE
clipper2/1.2.2: package(): Packaged 2 '.a' files: libClipper2.a, libClipper2Z.a
clipper2/1.2.2: package(): Packaged 7 '.h' files
clipper2/1.2.2: Created package revision a086408c84375b1cca07baf7f16bfcf6
clipper2/1.2.2: Package '38b07c1e6edb13b4efbe474d797049f16efb48bc' created
clipper2/1.2.2: Full package reference: clipper2/1.2.2#7725035548b161d11632f2647eee549b:38b07c1e6edb13b4efbe474d797049f16efb48bc#a086408c84375b1cca07baf7f16bfcf6
clipper2/1.2.2: Package folder /home/toge/.conan2/p/b/clipp5f826b8dae37d/p

======== Launching test_package ========

======== Computing dependency graph ========
Graph root
    clipper2/1.2.2 (test package): /home/toge/src/conan-center-index/recipes/clipper2/all/test_package/conanfile.py
Requirements
    clipper2/1.2.2#7725035548b161d11632f2647eee549b - Cache

======== Computing necessary packages ========
Requirements
    clipper2/1.2.2#7725035548b161d11632f2647eee549b:38b07c1e6edb13b4efbe474d797049f16efb48bc#a086408c84375b1cca07baf7f16bfcf6 - Cache

======== Installing packages ========
clipper2/1.2.2: Already installed! (1 of 1)

======== Testing the package ========
Removing previously existing 'test_package' build folder: /home/toge/src/conan-center-index/recipes/clipper2/all/test_package/build/gcc-13-x86_64-20-release
clipper2/1.2.2 (test package): Test package build: build/gcc-13-x86_64-20-release
clipper2/1.2.2 (test package): Test package build folder: /home/toge/src/conan-center-index/recipes/clipper2/all/test_package/build/gcc-13-x86_64-20-release
clipper2/1.2.2 (test package): Writing generators to /home/toge/src/conan-center-index/recipes/clipper2/all/test_package/build/gcc-13-x86_64-20-release/generators
clipper2/1.2.2 (test package): Generator 'VirtualRunEnv' calling 'generate()'
clipper2/1.2.2 (test package): Generator 'CMakeToolchain' calling 'generate()'
clipper2/1.2.2 (test package): CMakeToolchain generated: conan_toolchain.cmake
clipper2/1.2.2 (test package): CMakeToolchain generated: CMakePresets.json
clipper2/1.2.2 (test package): CMakeToolchain generated: ../../../CMakeUserPresets.json
clipper2/1.2.2 (test package): Generator 'CMakeDeps' calling 'generate()'
clipper2/1.2.2 (test package): Generating aggregated env files
clipper2/1.2.2 (test package): Generated aggregated env files: ['conanrun.sh', 'conanbuild.sh']

======== Testing the package: Building ========
clipper2/1.2.2 (test package): Calling build()
clipper2/1.2.2 (test package): Running CMake.configure()
clipper2/1.2.2 (test package): RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/home/toge/src/conan-center-index/recipes/clipper2/all/test_package/build/gcc-13-x86_64-20-release/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/home/toge/src/conan-center-index/recipes/clipper2/all/test_package" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/home/toge/src/conan-center-index/recipes/clipper2/all/test_package"
-- Using Conan toolchain: /home/toge/src/conan-center-index/recipes/clipper2/all/test_package/build/gcc-13-x86_64-20-release/generators/conan_toolchain.cmake
-- Conan toolchain: C++ Standard 20 with extensions OFF
-- The CXX compiler identification is GNU 13.1.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: Component target declared 'Clipper2::clipper2'
-- Conan: Component target declared 'Clipper2::clipper2z'
-- Conan: Target declared 'clipper2::clipper2'
-- Configuring done (0.1s)
-- Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_POLICY_DEFAULT_CMP0091


-- Build files have been written to: /home/toge/src/conan-center-index/recipes/clipper2/all/test_package/build/gcc-13-x86_64-20-release

clipper2/1.2.2 (test package): Running CMake.build()
clipper2/1.2.2 (test package): RUN: cmake --build "/home/toge/src/conan-center-index/recipes/clipper2/all/test_package/build/gcc-13-x86_64-20-release" -- -j12
[ 50%] Building CXX object CMakeFiles/test_package_z.dir/test_package_z.cpp.o
[ 50%] Building CXX object CMakeFiles/test_package.dir/test_package.cpp.o
[ 75%] Linking CXX executable test_package
[ 75%] Built target test_package
[100%] Linking CXX executable test_package_z
[100%] Built target test_package_z


======== Testing the package: Executing test ========
clipper2/1.2.2 (test package): Running test()
clipper2/1.2.2 (test package): RUN: ./test_package

clipper2/1.2.2 (test package): RUN: ./test_package_z
```